### ISSUE TITLE
Problems getting model to converge?

### ISSUE THREAD

**Issue by cjb06776** [tryolabs/luminoth#160]
Inspired by your blog post, wanted to give your toolkit a try and compare it to the tensorflow object detection API (https://github.com/tensorflow/models/tree/master/research/object_detection).   Have a simple data set with one class per image that I was using as my test case.

Other than adjusting the anchor box ratios (target class is much wider than tall), used the default sample config templates that are provided for luminoth and tensorflow object detection, with a resnet101 based model.   Tensorflow object detection gives decent results, but the luminoth model on the same data does not converge: loss explodes and goes to nan after processing only a handfull of images.  

```
INFO:tensorflow:step: 18, file: b'188.png', train_loss: 98180410638336.0, in 0.60s
INFO:tensorflow:step: 19, file: b'457.png', train_loss: 1.5213025840643598e+21, in 0.60s
INFO:tensorflow:step: 20, file: b'260.png', train_loss: 1.4238589805266597e+26, in 0.60s
INFO:tensorflow:step: 21, file: b'636.png', train_loss: nan, in 0.60s
```

I havn't really dug into what the differences between the two implementations are, was wondering if you have any insight into why this might have happened.   Since this converges fine out of the box with the tensorflow implementation, would have expected comparible results here.  Feels like something obvious I might have missed in setting up the config?  

Here is the config I used, the tf records were generated by converting from csv file ('filename.png, xmin, ymin, xmax, ymax, 1' format) using the dataset transform command.     

(for reference was using the faster_rcnn_resnet101_coco.config config template for tensorflow)

Any suggestions?

```
train:
  # Directory in which model checkpoints & summaries (for Tensorboard) will be saved
  job_dir: jobs/
  debug: True

dataset:
  type: object_detection
  # From which directory to read the dataset
  dir: datasets/voc/tf

model:
  type: fasterrcnn
  network:
    # Total number of classes to predict
    num_classes: 1

  base_network:
    # Which type of pretrained network to use
    architecture: resnet_v1_101
    # Should we train the pretrained network
    trainable: True
    # Should we download weights if not available
    download: True

  anchors:
    # Base size to use for anchors.
    base_size: 300
    # Scale used for generating anchor sizes.
    scales: [0.5, 1, 1.5]
    # Aspect ratios used for generating anchors.
    ratios: [.1, .4]
    # Stride depending on feature map size (of pretrained).
    stride: 16
```

**Comment by nagitsu** [comment]
Hello @cjb06776!

The current default configuration is not optimal, we'll be updating it shortly. For your use-case, you could try with the following (using the current `master`):

```train:
  run_name: single-class
  debug: True
  job_dir: jobs/

  learning_rate:
    _replace: True
    decay_method:
    learning_rate: 0.0003

dataset:
  dir: datasets/voc/tf
  
model:
  type: fasterrcnn
  network:
    num_classes: 1

  batch_norm: False

  base_network:
    architecture: resnet_v1_101
    fine_tune_from: block2
    output_stride: 16
    train_batch_norm: False
    use_tail: True

  anchors:
    scales: [0.25, 0.5, 1, 2]
    ratios: [.1, .4]

  rcnn:
    layer_sizes: []
    use_mean: True
```

This changes the last fully-connected layer with the last Resnet block, and it decreases the learning rate a bit (which is the probable cause of the loss explosion). Also, `base_size` should be left as-is if the `stride` is not modified.

Let me know if this works for you. If you run into any issues I could give it a try!

We're in the process of improving the documentation and having saner config defaults, but feel free to ask in the meantime. :smile:

**Comment by cjb06776** [comment]
Thanks for the suggestions!

Working much better now.  Although, I am still struggling to get the same quality of accuracy as the tensorflow object detection models (after training a comparable number of iterations, quite a few more false positive predictions).   

>  Also, base_size should be left as-is if the stride is not modified.

Could you elaborate on this?  What is the relatinoship between the anchor base size and the stride, how do you determine correct value if you do change it?   One can change the anchor scale/aspect ratio without changing the stride, but changing the base size does potentially impact other things?   


> ```   rcnn:
>   layer_sizes: [] 

What is the rational for removing all the layers in the rcnn component?   

thanks
